File: Sliver lV/1920.js
const fs = require("fs")
const input = fs.readFileSync("dev/stdin").toString().split("\n")

let Arr1 = [...input[1].split(" ").map(v => +v)]
let Arr2 = [...input[3].split(" ").map(v => +v)]

const set = new Set(Arr1)

let answer = Arr2.map(v => set.has(v) ? 1 : 0)

console.log(answer.join("\n"))

File: Sliver lV/1920.py
n = int(input())
a = list(map(int, input().split()))
m = int(input())
b = list(map(int,input().split()))

d = dict()

for i in a:
    d[i] = "1"

for i in b:
    if(d.get(i) == "1"):
        print(1)
    else:
        print(0)